### PR TITLE
Fix segmentation fault in AsString with large width values

### LINKED ISSUE #105292: Segmentation fault when using tf.raw_ops.AsString with large widthvalue
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf2.17

### Custom code

Yes

### OS platform and distribution

_No response_

### Mobile device

_No response_

### Python version

_No response_

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

A segmentation fault is triggered when calling tf.raw_ops.AsStringwith width=2147483647(INT_MAX). The crash occurs during execution, after GPU device initialization.

Expected Behavior:
The operation should either:
Successfully convert numbers to scientific notation strings with specified formatting, or
Return a clear error for invalid/unsupported parameter values
Actual Behavior:
Segmentation fault after GPU initialization logs.
Potential Cause:
Integer overflow or insufficient buffer allocation when processing extremely large width values.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf\ninput_data = tf.constant([3.1415926, 2.71828], dtype=tf.float32)\nresult = tf.raw_ops.AsString(input=input_data, precision=2, scientific=True,\n    shortest=False, width=2147483647, fill='0', name='asString')
```

### Relevant log output

```shell

```

### PR DESCRIPTION
**Pull request by ajayvnkt**
Add bounds checking to prevent segmentation faults when AsString is called with extremely large width parameter values (e.g., INT_MAX).

The fix adds validation to check for unreasonably large width values before attempting string formatting operations that would cause memory exhaustion and crashes.

Changes:
1. Add width bounds checking (max 1 million characters)
2. Return clear InvalidArgument error instead of crashing
3. Add comprehensive test coverage for edge cases

Fixes #105292 where width=INT_MAX causes segfault.

### CODE REVIEW

File: tensorflow/core/kernels/as_string_op.cc [hunk 1]
```diff
@@ -92,6 +92,16 @@ class AsStringOp : public OpKernel {
     if (width_ <= -1) {
       width_ = 0;
     }
+    
+    // Check for unreasonably large width values to prevent memory exhaustion
+    // and segmentation faults when absl::StrFormat tries to allocate buffers
+    const int kMaxReasonableWidth = 1000000;  // 1 million characters
```
Comment (mihaimaruseac) [inline tensorflow/core/kernels/as_string_op.cc:98]: Allocation with large numbers fails and returns a null pointer. Better to check for that than to use an arbitrary large constant which has no justification.

See also #105318 comments

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Looks like AI slop. You can use AI, but you need to clean the PR, make it do exactly what is says it does and nothing more. One commit per PR is enough, you don't need hundreds of commits to change just one file. Please don't waste reviewer time. Submitting more AI slop will get all of your future PRs desk rejected.

**Comment by mihaimaruseac** [comment]
Closing AI slop with no reaction from author after initial review.
